src/braket/experimental/autoqasm/errors.py

```diff
@@ -24,12 +22,28 @@ class UnknownQubitCountError(AutoQasmError):
     """Missing declaration for the number of qubits."""
 
     def __init__(self):
-        self.message = f"""Unspecified number of qubits.
+        self.message = """Unspecified number of qubits.
+
+Specify the number of qubits used by your program by supplying the \
+`num_qubits` argument to `autoqasm.function`. For example:
 
-Please declare the total number of qubits for your program. \
-You can do that by calling your AutoQASM program with the \
-{ProgramOptions.NUM_QUBITS.value} keyword argument. \
+    @autoqasm.function(num_qubits=5)
+    def my_autoqasm_function():
+        ...
 """
 
     def __str__(self):
         return self.message
+
+
+class InconsistentNumQubits(AutoQasmError):
+    """Num qubits supplied to main function does not match subroutine."""
+
+    def __init__(self):
+        self.message = """\
+The number of qubits specified by one of your subroutines does not match the \
+arguments supplied to your main function. Remove the num_qubits decorator \
+argument to all AutoQASM decorated functions except the one you wish to call."""
```

Comment:
"wish to call" can be confusing. They likely want to call them all (in some way or another) even if some of them are subroutines. 

Just needing this error points to that we'll likely making this in a way that's confusing.

Reply:
Updated the error message, thank you.

Yeah, it's a tough problem. So far, this is the least error prone, and best documented method that we've thought of that is currently available to us. At some point, we may expose two different kinds of decorators, or we may be able to perform transformations on the generated program, or we may support more output formats. If any of those changes are made, then we will have additional options for how to handle this.